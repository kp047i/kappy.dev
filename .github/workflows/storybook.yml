name: Storybook Tests

on:
  push:
  pull_request:

jobs:
  storybook:
    name: Storybook Tests
    runs-on: ubuntu-latest
    timeout-minutes: 60
    env:
      PLAYWRIGHT_BROWSERS_PATH: 0
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 22.13.0
      - name: Enable Corepack
        run: corepack enable
      - name: Install pnpm
        run: |
          corepack prepare pnpm@8.15.5 --activate
          pnpm --version
          which pnpm
      - name: Get pnpm store directory
        shell: bash
        run: |
          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
      - name: Setup pnpm cache
        uses: actions/cache@v4
        with:
          path: ${{ env.STORE_PATH }}
          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
          restore-keys: |
            ${{ runner.os }}-pnpm-store-
      - name: Install dependencies
        run: pnpm install --frozen-lockfile
      - name: Install Playwright browsers
        run: pnpm exec playwright install --with-deps chromium chromium-headless-shell
      - name: Build Storybook
        run: pnpm build-storybook --quiet
      - name: Storybook Vitest tests
        env:
          CI: true
        run: pnpm run test:storybook
