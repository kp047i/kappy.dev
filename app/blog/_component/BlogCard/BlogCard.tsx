import Link from "next/link";
import { Post } from "../../type";
import { BlogCategory } from "../BlogCategory/BlogCategory";
import { FaAngleRight } from "react-icons/fa6";

export type BlogCardProps = Pick<Post, "metadata">;

export function BlogCard({ metadata }: BlogCardProps) {
  return (
    <div className="space-y-4 transition-allrounded-xl">
      <div className="flex items-center space-x-4">
        <span className="text-3xl">{metadata.emoji}</span>
        <div className="space-y-2">
          <Link
            href={`/blog/${metadata.slug}`}
            className="text-xl font-semibold text-secondary-950 hover:underline"
          >
            {metadata.title}
          </Link>
          <p className="text-secondary-950 opacity-80">
            {metadata.publishedAt}
          </p>
        </div>
      </div>
      <p className="text-gray-700">{metadata.description}</p>
      <div className="flex items-end justify-between space-y-2 text-sm font-medium text-primary-700">
        <BlogCategory category={metadata.category} />
        <div>
          <Link href={`/blog/${metadata.slug}`} className="hover:underline">
            続きを読む
          </Link>
        </div>
      </div>
    </div>
  );
}
