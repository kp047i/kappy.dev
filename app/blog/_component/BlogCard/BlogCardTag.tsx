import Link from "next/link";
import { FaHashtag } from "react-icons/fa6";
export default function BlogCardTag({ tag }: { tag: string }) {
  return (
    <Link
      href={`/blog?tag=${tag}`}
      className="flex items-center px-2 py-1 space-x-1 transition-all border text-primary-700 border-primary-700 rounded-xl hover:bg-primary-700 hover:text-secondary-950"
    >
      <FaHashtag />
      <span>{tag}</span>
    </Link>
  );
}
