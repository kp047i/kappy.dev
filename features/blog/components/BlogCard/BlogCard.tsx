import Link from "next/link";

import { Post } from "../../type";
import { BlogCategory } from "../BlogCategory/BlogCategory";

export type BlogCardProps = Pick<Post, "metadata">;

export function BlogCard({ metadata }: BlogCardProps) {
  return (
    <div className="space-y-4 rounded-xl transition-colors duration-200">
      <div className="flex items-center space-x-4">
        <span className="text-3xl">{metadata.emoji}</span>
        <div className="space-y-2">
          <Link
            href={`/blog/${metadata.slug}`}
            className="text-xl font-semibold text-secondary-950 transition-colors hover:underline dark:text-base-50"
          >
            {metadata.title}
          </Link>
          <p className="text-secondary-950 opacity-80 dark:text-base-100">
            {metadata.publishedAt}
          </p>
        </div>
      </div>
      <p className="text-base-800 dark:text-base-100">{metadata.description}</p>
      <div className="flex items-end justify-between space-y-2 text-sm font-medium text-primary-700 dark:text-primary-300">
        <BlogCategory category={metadata.category} />
        <div>
          <Link
            href={`/blog/${metadata.slug}`}
            className="hover:underline dark:text-primary-300"
          >
            続きを読む
          </Link>
        </div>
      </div>
    </div>
  );
}
