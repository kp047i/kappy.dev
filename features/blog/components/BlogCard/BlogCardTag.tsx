import Link from "next/link";
import { FaHashtag } from "react-icons/fa6";
export default function BlogCardTag({ tag }: { tag: string }) {
  return (
    <Link
      href={`/blog?tag=${tag}`}
      className="flex items-center space-x-1 rounded-xl border border-primary-700 px-2 py-1 text-primary-700 transition-colors hover:bg-primary-700 hover:text-secondary-950 dark:border-primary-400 dark:text-primary-200 dark:hover:bg-primary-400 dark:hover:text-base-950"
    >
      <FaHashtag />
      <span>{tag}</span>
    </Link>
  );
}
