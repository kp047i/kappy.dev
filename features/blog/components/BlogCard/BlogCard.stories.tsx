import { Meta, StoryObj } from "@storybook/nextjs";
import { within, expect } from "storybook/test";

import { BlogCard, BlogCardProps } from "./BlogCard";

const meta: Meta<typeof BlogCard> = {
  component: BlogCard,
};

export default meta;
type Story = StoryObj<typeof BlogCard>;

const defaultArgs: BlogCardProps = {
  metadata: {
    title: "Title",
    description: "Description",
    emoji: "üìù",
    publishedAt: "2022-01-01",
    category: "tech",
    tags: ["react"],
    slug: "slug",
  },
};

export const Default: Story = {
  args: defaultArgs,
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    await expect(canvas.getByText(defaultArgs.metadata.title)).toBeVisible();
    await expect(canvas.getByText(defaultArgs.metadata.description)).toBeVisible();
    if (defaultArgs.metadata.emoji) {
      await expect(canvas.getByText(defaultArgs.metadata.emoji)).toBeVisible();
    }
    await expect(canvas.getByText(defaultArgs.metadata.publishedAt)).toBeVisible();

    const link = canvas.getByText("Á∂ö„Åç„ÇíË™≠„ÇÄ");
    expect(link).toHaveAttribute("href", `/blog/${defaultArgs.metadata.slug}`);
  },
};
