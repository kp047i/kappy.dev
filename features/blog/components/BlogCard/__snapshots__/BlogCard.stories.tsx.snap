// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog/components/BlogCard Default play-test 1`] = `
<div class="space-y-4 rounded-xl transition-colors duration-200">
  <div class="flex items-center space-x-4">
    <span class="text-3xl">
      üìù
    </span>
    <div class="space-y-2">
      <a class="text-xl font-semibold text-secondary-950 transition-colors hover:underline dark:text-base-50"
         href="/blog/slug"
      >
        Title
      </a>
      <p class="text-secondary-950 opacity-80 dark:text-base-100">
        2022-01-01
      </p>
    </div>
  </div>
  <p class="text-base-800 dark:text-base-100">
    Description
  </p>
  <div class="flex items-end justify-between space-y-2 text-sm font-medium text-primary-700 dark:text-primary-300">
    <a class="group inline-flex items-center gap-2 text-primary-700 opacity-80 transition-colors hover:opacity-100 dark:text-primary-300"
       href="/blog?category=tech"
    >
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 512 512"
           class="transform text-current group-hover:hidden"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
      >
        <path d="M448 480L64 480c-35.3 0-64-28.7-64-64L0 192l512 0 0 224c0 35.3-28.7 64-64 64zm64-320L0 160 0 96C0 60.7 28.7 32 64 32l128 0c20.1 0 39.1 9.5 51.2 25.6l19.2 25.6c6 8.1 15.5 12.8 25.6 12.8l160 0c35.3 0 64 28.7 64 64z">
        </path>
      </svg>
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 576 512"
           class="hidden transform text-current group-hover:block group-hover:m-0"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
      >
        <path d="M88.7 223.8L0 375.8 0 96C0 60.7 28.7 32 64 32l117.5 0c17 0 33.3 6.7 45.3 18.7l26.5 26.5c12 12 28.3 18.7 45.3 18.7L416 96c35.3 0 64 28.7 64 64l0 32-336 0c-22.8 0-43.8 12.1-55.3 31.8zm27.6 16.1C122.1 230 132.6 224 144 224l400 0c11.5 0 22 6.1 27.7 16.1s5.7 22.2-.1 32.1l-112 192C453.9 474 443.4 480 432 480L32 480c-11.5 0-22-6.1-27.7-16.1s-5.7-22.2 .1-32.1l112-192z">
        </path>
      </svg>
      <span>
        Tech
      </span>
    </a>
    <div>
      <a class="hover:underline dark:text-primary-300"
         href="/blog/slug"
      >
        Á∂ö„Åç„ÇíË™≠„ÇÄ
      </a>
    </div>
  </div>
</div>
`;
